### PR TITLE
Date Parse Bug Fixes

### PR DESCRIPTION
**Pull request by parc-jason**
SheetJS bugs - copy fix over into node_modules.
Add unit tests for Excel parse.
Tweak to Invalid Date handling - show original value.
Add Excel sample (same data as CSV).

Excel/CSV Library has some bugs around date parsing; we fix with a monkey patch.
When loading to NPM image, we load the existing library, then delete unnecessary files from the /node_modules, then copy over our fix.  You will see lint warnings on production buil - just ignore them.  I want to change the library code as little as possible.

Date/Timestamp fields were getting incorrect dates and times.  For example Oct 4 2019 (no hours/minutes) would come through as Oct 3 2019 (actually Oct 3 2019 4pm). The library wasn't handling differences in timezones.  
Also, wanted to show original data and not 'Invalid Date' when we fields were thought to be dates. 

[SHOWCASE-614](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-614)
[SHOWCASE-625](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-625)
## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments
We should keep an eye on sheetjs, when fixes are made, we can just import their library.

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-z208i4-tools.pathfinder.gov.bc.ca/dashboard?id=nr-messaging-service-showcase-pr-85

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-z208i4-tools.pathfinder.gov.bc.ca/dashboard?id=nr-messaging-service-showcase-pr-85

File: frontend/src/utils/ExcelUtils.test.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    //endDateTS[1] = Oct 6 2019 13:45
+    expect(contexts[1].context.endDateTs).toBe('2019-10-06 13:45');
+    //endDateTS[2] = 6 Oct 2019 13:45
+    expect(contexts[2].context.endDateTs).toBe('2019-10-06 13:45');
+    //endDateTS[3] = 10-6-19 13:45:00
+    expect(contexts[3].context.endDateTs).toBe('2019-10-06 13:45');
+    //endDateTS[4] = Oct 6 2019 1:45:00 PM
+    expect(contexts[4].context.endDateTs).toBe('2019-10-06 13:45');
+  });
+  it('handles bad date datacorrectly', async () => {
```
Comment (jujaga) [inline frontend/src/utils/ExcelUtils.test.jsx]: space between data and correctly

File: openshift/frontend-npm.bc.yaml [hunk 2]
```diff
@@ -45,10 +45,22 @@ objects:
...
+        # so we will manually copy this file over here
+        # remove stuff we do not need, add the file we want...
+        RUN rm -rf /opt/app-root/src/node_modules/xlsx/bin && \
+          rm /opt/app-root/src/node_modules/xlsx/dist/*.min.js && \
+          rm /opt/app-root/src/node_modules/xlsx/dist/*.min.map && \
+          rm /opt/app-root/src/node_modules/xlsx/dist/*.extendscript.js && \
+          rm /opt/app-root/src/node_modules/xlsx/dist/xlsx.js && \
+          rm /opt/app-root/src/node_modules/xlsx/xlsx.mini.js
+        COPY xlsx-override.js /opt/app-root/src/node_modules/xlsx/xlsx.js
+        COPY xlsx-override.js /opt/app-root/src/node_modules/xlsx/dist/xlsx.js
```
Comment (jujaga) [inline openshift/frontend-npm.bc.yaml]: Just as a note, you can use the following in the dockerfile to reduce redundancy
```
ARG APP_ROOT=/opt/app-root/src
```
Then you can have that variable be replaced across the entire dockerfile
```
WORKDIR ${APP_ROOT}
...
  rm ${APP_ROOT}/node_modules/xlsx/xlsx.mini.js
```

File: frontend/src/ches/MergeAbout.jsx [hunk 3]
```diff
@@ -26,10 +26,17 @@ class MergeAbout extends Component {
         <p>To showcase CHES, we have added some nice features to turn an Excel spreadsheet or CVS file into a
           CHES Mail Merge request. See below for some sample data and a template you can use as a guide to
           creating your own data and templates. It is advised that you fix the data in your spreadsheet and not
-          in the JSON editor.</p>
+          in the JSON editor.
+        </p>
+        <p>
+          The Excel spreadsheet/CVS file parser does best guesses on dates and times, it cannot parse and understand all
+          formats.  If you encounter difficulties with dates and times from Excel, convert your sheet to CSV and format
+          your dates to &apos;YYYY-MM-DD&apos; and time stamps to &apos;YYYY-MM-DD HH:mm&apos;.
```
Comment (loneil) [inline frontend/src/ches/MergeAbout.jsx]: Can use quote here right? Don't need &apos;

**Pull request merged**